src/dataset-builder/DatasetBuilder.test.ts

```diff
@@ -143,6 +143,23 @@ describe('DatasetBuilder', () => {
       })
     );
 
+  const mockCreateSnapshotAccessRequest = jest.fn().mockResolvedValue({
+    id: '',
+    datasetId: null,
```

Comment:
Did you think about omitting `datasetId` here? It should resolve to `undefined`, which would make things easier when we pull that field later.

Reply:
I didn't think about omitting datasetId, let me try that